src/addons/zigbee-adapter/zb-classifier.js

```diff
@@ -10,22 +10,73 @@
 
 'use strict';
 
-var zclId = require('zcl-id');
-var utils = require('../utils');
-var ZigBeeProperty = require('./zb-property');
+const zclId = require('zcl-id');
+const utils = require('../utils');
+const ZigBeeProperty = require('./zb-property');
 
 const ZHA_PROFILE_ID = zclId.profile('HA').value;
 const ZHA_PROFILE_ID_HEX = utils.hexStr(ZHA_PROFILE_ID, 4);
 const CLUSTER_ID_GENONOFF = zclId.cluster('genOnOff').value;
 const CLUSTER_ID_GENONOFF_HEX = utils.hexStr(CLUSTER_ID_GENONOFF, 4);
+const CLUSTER_ID_GENLEVELCTRL = zclId.cluster('genLevelCtrl').value;
+const CLUSTER_ID_GENLEVELCTRL_HEX = utils.hexStr(CLUSTER_ID_GENLEVELCTRL, 4);
 
 const THING_TYPE_ON_OFF_SWITCH = 'onOffSwitch';
+const THING_TYPE_MULTI_LEVEL_SWITCH = 'multiLevelSwitch';
 
 class ZigBeeClassifier {
 
   constructor() {
   }
 
+  addLevelProperty(node, endpoint) {
+    this.addProperty(
+      node,                           // device
+      'level',                        // name
+      {                               // property description
+        type: 'number',
+        unit: '%',
+        min: 0,
+        max: 100,
+      },
+      ZHA_PROFILE_ID,                 // profileId
+      endpoint,                       // endpoint
+      CLUSTER_ID_GENLEVELCTRL,        // clusterId
+      'currentLevel',                 // attr
+      'setLevelValue',                // setAttrFromValue
+      'parseLevelAttr'                // parseValueFromAttr
+    );
+  }
+
+  addOnProperty(node, endpoint) {
+    this.addProperty(
+      node,                           // device
+      'on',                           // name
+      {                               // property description
+        type: 'boolean',
+      },
+      ZHA_PROFILE_ID,                 // profileId
+      endpoint,                       // endpoint
+      CLUSTER_ID_GENONOFF,            // clusterId
+      'onOff',                        // attr
+      'setOnOffValue',                // setAttrFromValue
+      'parseOnOffAttr'                // parseValueFromAttr
+    );
+  }
+
+  addProperty(node, name, descr, profileId, endpoint, clusterId,
+              attr, setAttrFromValue, parseValueFromAttr) {
+    let property =  new ZigBeeProperty(node, name, descr, profileId,
```

Comment:
nit: Could we change all instances of ZigBee to Zigbee? Zigbee is the proper name.

Reply:
Huh, you're right. I must have been getting this wrong for years! Very commonly capitalized ZigBee, but the Zigbee Alliance website doesn't capitalize the B. Wonder if they changed it at some point...